api/src/index.js

```diff
@@ -11,21 +11,97 @@ app.use(bodyParser.json());
 
 const apiRouter = express.Router();
 
-// You can delete this route once you add your own routes
-apiRouter.get("/", async (req, res) => {
-  const SHOW_TABLES_QUERY =
-    process.env.DB_CLIENT === "pg"
-      ? "SELECT * FROM pg_catalog.pg_tables;"
-      : "SHOW TABLES;";
-  const tables = await knex.raw(SHOW_TABLES_QUERY);
-  res.json({ tables });
-});
+//future-meals
+apiRouter.get("/future-meals", async (req, res) => {
+  try {
+    const futureMeals = await knex.raw("Select * From Meal WHERE Meal.when > NOW()")
+    if (futureMeals[0].length === 0) {
+    return res.status(404).json("There are no meals for the future.");
+    }
+    res.json(futureMeals[0]);
+  } catch (error) {
+    console.error(error)
+    res.status(500).json('Server error!')
+  }
+})
+//past-meals
+apiRouter.get("/past-meals", async (req, res) => {
+  try {
+    const pastMeals = await knex.raw("Select * From Meal WHERE Meal.when < NOW()") 
+  if(pastMeals[0].length === 0) {
+    return res.status(404).json("There are no meals in the past.")
+  }
+  res.json(pastMeals[0]);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json('Server error')
+  }
+})
+//all-meals
+apiRouter.get("/all-meals" , async (req, res) =>{
+  try {
+    const meals = await knex.raw("SELECT * FROM Meal ORDER BY id DESC");
+    if(meals[0].length === 0) {
+      return res.status(404).json('There are no meals!')
+    }
+    res.json(meals[0]);
+
+  } catch (error) {
+    console.log(error)
+    res.status(500).json('Server error')
+  }
+ 
+})
+//first-meal
+apiRouter.get("/first-meal", async (req, res)=>{
+  try {
+    const firstMeal = await knex.raw("SELECT * FROM Meal ORDER BY id ASC LIMIT 1");
+    if(firstMeal[0].length===0){
+      return res.status(404).json("No meals found!")
+    }
+    res.json(firstMeal[0][0])
```

Comment:
A small suggestion is to use some variables to make the code more readable. Looking at `firstMeal[0][0]` is a bit confusing. You can make this easier to understand by having something like:
```javascript
const result = await knex.raw(...);
const meals = result[0];
const firstMeal = meals[0];
```